### ISSUE TITLE
Name-space classes

### ISSUE THREAD

**Issue by sprynm** [PrismJS/prism#962]
# While the concept of "sensible class names" is fantastic compared to esoteric "abbreviated nonsense" ... "tag", "comment" and "property" etc. are awfully generic terms.

Increasing CSS specificity is a bad path so is there a chance of creating a safe namespace through the build process?  

Something like `.prism-token .prism-token-tag` would be ideal.  gzip will take care of any file size increase.

Anyone looked into this?

EDIT: #947 seems to address this but with more client side code.  I'd much prefer the build handle the classnames

**Comment by sprynm** [comment]
Did a bunch of reading the issue history and it seems this is a touchy subject.  Key limitation now seems to be the backward compatibility so any solution would have to be opt-in only.  I don't see it being THAT big of a deal.  Potential steps:

1) Decided the best place to set the constant so it can be easily accessed when built/run.
2) Set the constant to "" so the build is backward compatible by default.
3) Adjust prism-core.js, line 426 to something like:
```
	var namespace = _.constants.namespace ? _.constants.namespace+"-" : "";
        var env = {
                ...
		classes: [ namespace + 'token', namespace + o.type],
                ...
        }
```
4) create a SASS/LESS variable to automatically update themes
5) Use a environment flag in the gulpfile to customize the build

Now running something like `gulp --namespace=Wookie` builds a custom set of files with zero collisions.

Perhaps on the page something like `Prism.constants.namespace = 'Wookie';` could set the variable on the fly?  CSS is up to the author then.

I don't think the language packs need to be updated in this scenario?

Disclaimer, this is only after my first run through the source code so I may have misunderstood the structure badly.  Apologies if wasting your time.

**Comment by zeitgeist87** [comment]
There is a PR for a plugin that changes the class names at runtime #950. Would that be an acceptable solution?

**Comment by lukewhitehouse** [comment]
Hey!

Whats the likelihood of this getting implemented? Unfortunately I ran into this problem today as I had already used `.comment` to style up a Comment listing, something I imagine will be very common for blogs.

Just seems a shame that I've now had to namespace my own comments section rather than a plugin doing this.

**Comment by LeaVerou** [comment]
The PR @zeitgeist87 linked to above is actually merged now, so I'm closing this. Please use that plugin. If you want to make suggestions for that plugin's functionality, please open a new issue and tag @dvkndn

**Comment by lukewhitehouse** [comment]
Cheers @LeaVerou :)
